samples/repo-search/desktopApp/build.gradle.kts

```
@@ -0,0 +1,35 @@
+import org.jetbrains.compose.desktop.application.dsl.TargetFormat
+
+plugins {
+  kotlin("multiplatform") // kotlin("jvm") doesn't work well in IDEA/AndroidStudio (https://github.com/JetBrains/compose-jb/issues/22)
+  id("org.jetbrains.compose")
+}
+
+kotlin {
+  jvm {
+    withJava()
+  }
+
+  sourceSets {
+    named("jvmMain") {
```

```suggestion
    val jvmMain by getting {
```
I actually hate this `by getting` nonsense, but just want to maintain consistency with other build scripts

---

I've actually been wondering what's the difference between:
`named("jvmMain")`, `val jvmMain by getting { .... }` and `getByName("jvmMain")` and `val jvmMain by creating { .... }`. 

But agreed, will update so it's consistent at least

---

First three are equivalent. Last one creates a source set instead of lookup up an existing one (also doable with an imperative API like `create(String)`, or something like that).

---

Nice - thanks for the explanation. I'll leave this thread unresolved in case anyone else finds this insight helpful.